eng/tools/CreateBaselineUpdatePR/PRCreator.cs

```diff
@@ -40,29 +40,62 @@ public async Task<int> ExecuteAsync(
 
         var updatedTestsFiles = GetUpdatedFiles(updatedFilesDirectory);
 
-        // Create a new tree for the originalFilesDirectory based on the target branch
-        var originalTreeResponse = await _client.Git.Tree.GetRecursive(_repoOwner, _repoName, targetBranch);
-        var testResultsTreeItems = originalTreeResponse.Tree
-            .Where(file => file.Path.Contains(originalFilesDirectory) && file.Path != originalFilesDirectory)
-            .Select(file => new NewTreeItem
-            {
-                Path = Path.GetRelativePath(originalFilesDirectory, file.Path),
-                Mode = file.Mode,
-                Type = file.Type.Value,
-                Sha = file.Sha
-            })
-            .ToList();
+        // Fetch the files within the desired path from the original tree
+        TreeResponse originalTreeResponse = await _client.Git.Tree.Get(_repoOwner, _repoName, targetBranch);
+        List<NewTreeItem> originalTreeItems = await FetchOriginalTreeItemsAsync(originalTreeResponse, targetBranch, originalFilesDirectory);
 
         // Update the test results tree based on the pipeline
-        testResultsTreeItems = await UpdateAllFilesAsync(updatedTestsFiles, testResultsTreeItems, pipeline);
-        var testResultsTreeResponse = await CreateTreeFromItemsAsync(testResultsTreeItems);
+        originalTreeItems = await UpdateAllFilesAsync(updatedTestsFiles, originalTreeItems, pipeline);
+        var testResultsTreeResponse = await CreateTreeFromItemsAsync(originalTreeItems);
         var parentTreeResponse = await CreateParentTreeAsync(testResultsTreeResponse, originalTreeResponse, originalFilesDirectory);
 
         await CreateOrUpdatePullRequestAsync(parentTreeResponse, buildId, title, targetBranch);
 
         return Log.GetExitCode();
     }
 
+    private async Task<List<NewTreeItem>> FetchOriginalTreeItemsAsync(
+        TreeResponse? treeResponse,
+        string targetBranch,
+        string desiredPath,
+        string relativePath = "")
+    {
+        var treeItems = new List<NewTreeItem>();
```

Comment:
I prefer this syntax because it's shorter.

```suggestion
        List<NewTreeItem> treeItems = [];
```

Reply:
Addressed in https://github.com/dotnet/dotnet/pull/331/commits/4d5896a56ed2026cc85abf490c34a60b1a8ae55b